evm/src/cpu/kernel/asm/hash/sha2/message_schedule.asm

```diff
@@ -3,32 +3,35 @@
     // stack: num_blocks
     %mul_const(64)
     %add_const(2)
+    PUSH @SEGMENT_KERNEL_GENERAL
+    GET_CONTEXT
+    %build_address
 %endmacro
 
 // Precodition: stack contains address of one message block, followed by output address
```

Comment:
Nit, not exactly part of the PR but since you corrected the typo in main.asm:
```suggestion
// Precondition: stack contains address of one message block, followed by output address
```